docs/configuration.rst

```diff
@@ -78,49 +78,46 @@ The JBoss EAP artifact will be fetched from: ``http://cache.host.com/cache/jboss
 
 .. code::
 
-    [repositories]
-    jboss-ocp=http://host/jboss-rhel-os.repo,http://host/jboss-rhel-ocp.repo,http://host/jboss-rhel-rhscl.repo
-    other-repos=http://otherhost.com/osme.repo
+   [repositories-$builder]
+   $repo_name = $action
 
-In case you have YUM/DNF repo files that you want to put to the ``/etc/yum.repos.d`` directory to enable additional
-repositories Cekit can handle it for you automatically.
+Where:
 
-Repositories are defined using custom keys that could be later referenced in the :ref:`image descriptor's
-packages section <descriptor_packages>`, under ``repositories``.
+* ``$builder`` - one of supported builder (docker, osbs)
 
-.. code:: yaml
+* ``$repo_name`` - name of the repository
+
+* ``$action`` - one of the following actions:
+
+  1) **odcs-pulp** - default for OSBS builder
 
-    packages:
-        repositories:
-            - jboss-ocp
-        install:
-            - mongodb24-mongo-java-driver
-            - postgresql-jdbc
-            - mysql-connector-java
-            - maven
-            - hostname
+     Aks odcs to create a temporary repository from an existing pulp repository. Repository name is taken from
+     ``repository`` atribute of the repository section.
 
-Cekit will copy all repo files to ``/etc/yum.repos.d`` directory and
-enable them to be used while installing packages listed in the packages section.
+  2) **rpm**
 
-At the end of the image build process Cekit removes newly added repo files from the ``/etc/yum.repos.d``
-directory automatically. If you do not want to have these files removed after installation --
-you need to make your repo files part of some module that installs them in the correct place.
+     Install repository from rpm via yum. Mostly used for epel/scl in Centos. RPM name is taken from ``repository``
+     attribute of the repository section.
 
-There are a few rules about repositories added that way:
+  3) **dummy** - default for Docker builder
+
+     Threats the repository as only documentation note, that this repo should be available in the image. It
+     does not inject/enable it in the image.
+
+*Example*: To define that Software collection repository will be installed via RPM for Docker builder and injected via ODCS pulp for OSBS you need to put following lines into you ``~/.cekit/config`` file:
+
+.. code::
 
-1. This feature covers only the situation where you want to add a custom repo file at build time but you do not want it to be enabled in containers.
-2. Repo file name should be the same as the repo id in the repository (the name between square brackets).
-3. There should be only one repository per file.
-4. Only added repositories will be enabled during install of packages, all other repositories (including default) will be disabled.
+   [repositories-docker]
+   scl = rpm
 
+   [repositories-osbs]
+   scl = odcs-pulp
```

Comment:
All documentation available lacks some examples that would have everything needed in a single place. It's hard to switch between documentation pages to come up with a solution that would work. I suggest extending the example. Currently it's very hard to track what is what what if this is a required keyword or something we need to come up with. Or how it relates to the descriptor section.